kolibri/core/content/management/commands/generate_schema.py

```diff
@@ -31,6 +32,10 @@ def add_arguments(self, parser):
         parser.add_argument('version', type=str)
 
     def handle(self, *args, **options):
+        # Restrict to Python 3 due to issues with `json.dump` on Python 2
```

Comment:
what sort of issues?

Reply:
```
(kolibri)  boni@t470s  ~/projects/learning-equality/kolibri  (feature/optimize-contentnode-endpoint) >  kolibri manage generate_schema 3
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
/home/boni/.venvs/kolibri/local/lib/python2.7/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
Installed 25 object(s) from 1 fixture(s)
Traceback (most recent call last):
  File "/home/boni/.venvs/kolibri/bin/kolibri", line 9, in <module>
    load_entry_point('kolibri', 'console_scripts', 'kolibri')()
  File "/home/boni/projects/learning-equality/kolibri/kolibri/utils/cli.py", line 631, in main
    manage(command, args=django_args)
  File "/home/boni/projects/learning-equality/kolibri/kolibri/utils/cli.py", line 421, in manage
    execute_from_command_line(argv=argv)
  File "/home/boni/projects/learning-equality/kolibri/kolibri/dist/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/boni/projects/learning-equality/kolibri/kolibri/dist/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/boni/projects/learning-equality/kolibri/kolibri/dist/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/boni/projects/learning-equality/kolibri/kolibri/dist/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/boni/projects/learning-equality/kolibri/kolibri/core/content/management/commands/generate_schema.py", line 65, in handle
    json.dump(data, f)
  File "/usr/lib/python2.7/json/__init__.py", line 190, in dump
    fp.write(chunk)
TypeError: write() argument 1 must be unicode, not str
```

Reply:
@indirectlylit FYI I discussed this with @rtibbles and we agreed that since this is internal command that it might be easiest to require Python 3 - but if you feel we should support Python 2 as well here I can play with this to figure out something that works?

Reply:
I found this SO answer helpful earlier: https://stackoverflow.com/a/36008538/6817428

Reply:
Main issue is that currently builds _only_ work in py2, so that leaves us with an inconsistent dev experience: https://github.com/learningequality/kolibri/blob/develop/Makefile#L122

@benjaoming do you recall why py2 is necessary for building?

Reply:
Yes, we only build on Python 2. That's the only way currently to have a Py2+3 build.

So we have to have everything py2+3, there is no nice way to handle exceptions to the rule IMO.

Reply:
I mean, it's not like I changed any previous behaviour here, this wasn't working on Python 2 anyway, I just added a message to the user to avoid them having to figure out what's going on.

Reply:
Oh right okay, then let's just say that someone at some point in time will have to find out that this `generate_schema` doesn't work on python 2 and fix that if it's a problem :)

Reply:
Yep, I'll try to take a look at it within this PR, AFAIK there is a difference between using `json.dumps()` and `json.dump()` with respect to the `ensure_ascii` (among other things) so there might be an easy solution.

Reply:
FYI: I removed this commit from this PR and created a separate PR with a compatibility fix: https://github.com/learningequality/kolibri/pull/4020